### PR TITLE
Jetpack to 10.5

### PR DESCRIPTION
**Pull request by jesse-greathouse**
## Description
* Changes the default version of Jetpack to 10.5
* Adds  backwards compatibility logic that prevents the user from using Jetpack 10.5 unless they are on Wordpress 5.8+

### Plugin Updated: Jetpack 10.5
The Default version of Jetpack is 10.5, in mu-plugins.

## Checklist

- [x] This change works and has been tested locally (or has an appropriate fallback).
- [x] This change works and has been tested on a Go sandbox.
- [ ] This change has relevant unit tests (if applicable).
- [ ] This change has relevant documentation additions / updates (if applicable).
- [x] I've created a changelog description that aligns with the provided examples.

## Steps to Test
1. Check out PR.
1. Go to `wp-admin` > `Jetpack`
1. Verify Jetpack is 10.5

### CODE REVIEW

File: jetpack.php [hunk 1]
```diff
@@ -22,8 +22,10 @@
 		define( 'VIP_JETPACK_DEFAULT_VERSION', '9.4' );
 	} elseif ( version_compare( $wp_version, '5.7', '<' ) ) {
 		define( 'VIP_JETPACK_DEFAULT_VERSION', '9.8' );
-	} else {
+	} elseif ( version_compare( $wp_version, '5.8', '<' ) ) {
```
Comment (rinatkhaziev) [inline jetpack.php:25]: I think this should be `5.9`?

Comment (rinatkhaziev) [reply jetpack.php:25]: nevermind that was wrong :)

**Review by rinatkhaziev — APPROVED** [review]

**Comment by jesse-greathouse** [comment]
r-2188

**Comment by jesse-greathouse** [comment]
Release v20220113.1

**Pull request merged**